airflow/api_fastapi/core_api/datamodels/pools.py

```diff
@@ -72,6 +72,25 @@ class PoolPatchBody(BaseModel):
 class PoolPostBody(BasePool):
     """Pool serializer for post bodies."""
 
-    pool: str = Field(alias="name")
+    pool: str = Field(alias="name", max_length=256)
     description: str | None = None
     include_deferred: bool = False
+
+
+class PoolPostBulkBody(BaseModel):
+    """Pools serializer for post bodies."""
+
+    pools: list[PoolPostBody]
```

Comment:
Do we need to raise error when duplicates are given? I can think of using `set[PoolPostBody]` to automatically handle this. WDYT?

Reply:
Sounds reasonable, I will refactor with set[PoolPostBody], thanks !

Reply:
Hi @rozza,  

While refactoring `pools` with `set`, I recalled why I initially chose not to use `set` instead of `list` with `field_validator`. Our primary goal is to ensure that `PoolPostBody.name` is unique within the list. However, using a `set` on whole  `PoolPostBody` leads to unhashable type issues in Pydantic.  

```
E   TypeError: unhashable type: 'PoolPostBody'
```

Reply:
If our primary goal is to ensure name to be unique, maybe we can modify PoolPostBody to include `__hash__`?

```
    def __hash__(self):
        return hash(self.pool)
```

Maybe its simpler in the way you already implemented. Otherwise, we also need to make a note that the first unique pool name will be considered and rest will be discarded

@pierrejeambrun , @amoghrajesh  wdyt?

Reply:
I'm not familiar with this `PoolPostBody` 🤔 . But if it's an immutable thing, I think implementing a `__hash__` should be fine

Reply:
I would be for letting the db handle that. The constraint is already enforce there, and we don't need it for the application code.